src/model/clients/GenericResponseError.ts

```diff
@@ -0,0 +1,90 @@
+import { StatusCodes } from 'shared/constants';
+
+import { IHttpResponseConfig } from './GenericResponse';
+import { HttpBody } from './HttpClient';
+
+export class GenericResponseError<T = HttpBody> extends Error {
```

Comment:
Necessary so we could throw actual `Error` objects when requests fail. Otherwise we get pretty bad stack traces in Sentry.